sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/PagedResponse.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.http.models;
+
+import io.clientcore.core.util.binarydata.BinaryData;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Response of a REST API that returns page.
+ *
+ * @see Response
+ *
+ * @param <T> The type of items in the page.
+ */
+public final class PagedResponse<T> implements Response<List<T>> {
```

Comment:
For azure-core, the `PagedResponse` is an interface. The implementation is in `PagedResponseBase`. Should we do same for clientcore?

In such case, we can even generate the implementation class in emitter (so clientcore does not need to have the impl class).